includes/Model/Story.php

```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Class Story
+ *
+ * @package   Google\Web_Stories\Model
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Model;
+
+use Google\Web_Stories\Story_Post_Type;
+use Google\Web_Stories\Media;
+use WP_Post;
+
+/**
+ * Class Story
+ *
+ * @package Google\Web_Stories\Model
+ */
+class Story {
+	/**
+	 * Title.
+	 *
+	 * @var string
+	 */
+	protected $title = '';
+	/**
+	 * URL.
+	 *
+	 * @var string
+	 */
+	protected $url = '';
+	/**
+	 * Markup.
+	 *
+	 * @var string
+	 */
+	protected $markup = '';
+	/**
+	 * Poster url - portrait.
+	 *
+	 * @var string
+	 */
+	protected $poster_portrait;
+	/**
+	 * Poster url - landscape.
+	 *
+	 * @var string
+	 */
+	protected $poster_landscape;
+	/**
+	 * Poster url - square.
+	 *
+	 * @var string
+	 */
+	protected $poster_square;
+
+	/**
+	 * Story constructor.
+	 *
+	 * @param Array $story Array of attributes.
+	 */
+	public function __construct( array $story = [] ) {
+		if ( ! empty( $story ) && is_array( $story ) ) {
+			foreach ( $story as $key => $value ) {
+				$this->$key = $value;
+			}
+		}
```

Comment:
```suggestion
		foreach ( $story as $key => $value ) {
			if ( property_exists( $this, $key ) ) {
				$this->$key = $value;
			}
		}
```

The `is_array` check is unnecessary because of the typing.

The `empty` check is unnecessary because the loop will do nothing for empty arrays

`property_exists` to ensure no arbitrary data is added.

Reply:
Fixed